SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -333,9 +355,9 @@ - (SDLStreamHasBytesHandler)sdl_controlStreamHasBytesHandlerForAccessory:(EAAcce
 
             // Determine protocol string of the data session, then create that data session
             NSString *indexedProtocolString = [NSString stringWithFormat:@"%@%@", indexedProtocolStringPrefix, @(buf[0])];
-            dispatch_sync(dispatch_get_main_queue(), ^{
+            if (accessory.isConnected){
```

Comment:
There is a race condition being guarded against here that was found during debugging. Since the handler is invoked on the receive dispatch queue, a disconnect can occur *underneath* the queue dispatching the block causing the data to be sent to the transport when it is in the process of being torn down.